docs/src/starknet/calldata-transformation.md

```diff
@@ -65,100 +65,118 @@ $ sncast --account myuser \
+    --arguments '0x10, 3, data_stransformer_contract::Enum::One' \
     --block-id latest
 ```
 
 getting the same result.
+Note that the arguments must be:
 
-> 📝 **Note**
-> All data types are serialized according to the official [Starknet specification](https://docs.starknet.io/architecture-and-concepts/smart-contracts/serialization-of-cairo-types/).
+* provided as a single string
+* comma (`,`) separated
 
 > 📝 **Note**
 > User-defined items such as enums and structs should be referred to depending on a way they are defined in ABI.\
 > In general, paths to items have form: `<project-name>::<module-path>::<item-name>`.
 
 ## Supported Expressions
 
-> 💡 **Info**
-> Only **constant** expressions are supported. Defining and referencing variables and calling functions (either builtin, user-defined or external) is not allowed.
-
 Cast supports most important Cairo corelib types:
-  * `bool`
-  * signed integers (`i8`, `i16`, `i32`, `i64`, `i128`)
-  * unsigned integers (`u8`, `u16`, `u32`, `u64`, `u96`, `u128`, `u256`, `u384`, `u512`)
-  * `felt252` (numeric literals and so-called *shortstrings*)
-  * `ByteArray`
-  * `ContractAddress`
-  * `ClassHash`
-  * `StorageAddress`
-  * `EthAddress`
-  * `bytes31`
-  * `Array` - using `array![]` macro
 
-Numeric types (primitives and `felt252`) can be paseed with type suffix specified -\
-for example `--calldata (420_u64,)`.
+* `bool`
+* signed integers (`i8`, `i16`, `i32`, `i64`, `i128`)
+* unsigned integers (`u8`, `u16`, `u32`, `u64`, `u96`, `u128`, `u256`, `u384`, `u512`)
+* `felt252` (numeric literals and `'shortstrings'`)
+* `ByteArray`
+* `ContractAddress`
+* `ClassHash`
+* `StorageAddress`
+* `EthAddress`
+* `bytes31`
+* `Array` - using `array![]` macro
+
+Numeric types (primitives and `felt252`) can be paseed with type suffix specified for example `--arguments 420_u64`.
+
+> 📝 **Note**
+> Only **constant** expressions are supported. Defining and referencing variables and calling functions (either builtin,
+> user-defined or external) is not allowed.
 
-## More Complex Examples
+### More Complex Examples
 
-  1. `complex_fn` - different data types:
+1. `complex_fn` - different data types:
 
   ```shell
-  $ sncast --account myuser \
-      call \
+  $ sncast call \
       --url http://127.0.0.1:5050 \
       --contract-address 0x016ad425af4585102e139d4fb2c76ce786d1aaa1cfcd88a51f3ed66601b23cdd \
       --function complex_fn \
-      --calldata (\
-          array![array![1, 2], array![3, 4, 5], array![6]], \
-          12, \
-          -128_i8, \
-          "Some string (a ByteArray)", \
-          ('a shortstring', 32_u32), \
-          true, \
-          0xffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff \
-      ) \
+      --arguments \
+'array![array![1, 2], array![3, 4, 5], array![6]],'\
+'12,'\
+'-128_i8,'\
+'"Some string (a ByteArray)",'\
+"('a shortstring', 32_u32),"\
+'true,'\
+'0xffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff' \
```

Comment:
nit:
Actually in this form (single quotes each line + bakslash) indentations and whitespaces seem to work correctly. If this is true, it would improve the readability of this example. Also the note below could be moved further down to highlight its relevance primarily for cases where a pair of single quotes is expanded on multiple lines.

Reply:
I'll verify

Reply:
Seems to break with indents, I'll leave it as is.